run_dir/static/js/running_notes.js

```diff
@@ -7,26 +7,30 @@ $(function(){
     });
 });
 
-function generate_category_label(category){
-    if (category == 'Workset'){
-         category='<span class="label label-primary">'+ category +"</span>";
-     }else if (category == 'Flowcell'){
-         category='<span class="label label-success">'+ category +"</span>";
-     }else if (category == 'Meeting' || category == "Decision"){
-         category='<span class="label label-info">'+ category +"</span>";
-     }else if (category == 'User Communication'){
-         category='<span class="label label-danger">'+ category +"</span>";
-     }else if (category == 'Bioinformatics'){
-         category='<span class="label label-warning">'+ category +"</span>";
-     }else if (category == 'Important'){
-         category='<span class="label label-imp">'+ category +"</span>";
-     }else if (category == 'Deviation'){
-	       category='<span class="label label-devi">'+ category +"</span>";
-     }else if (category != ''){
-         category='<span class="label label-default">'+ category +"</span>";
-     }
-     return category;
+function generate_category_label(cat_label){
+     var cat_classes = {
+        'Workset': ['primary', 'calendar-plus'],
+        'Flowcell': ['success', 'grip-vertical'],
+        'Decision': ['info', 'thumbs-up'],
+        'Lab': ['success', 'bong'],
+        'Bioinformatics': ['warning', 'laptop-code'],
+        'User Communication': ['usr', 'people-arrows'],
+        'Administration': ['danger', 'folder-open'],
+        'Important': ['imp', 'exclamation-circle'],
+        'Deviation': ['devi', 'frown']
+    }
+    // Remove the whitespace
+    var cat_label = cat_label.trim()
+    // Check if we recognise this category in the class object keys
+    if (Object.keys(cat_classes).indexOf(cat_label) != -1){
+        cat_label = '<span class="label label-'+ cat_classes[cat_label][0] +'">'+ cat_label + '&nbsp;' + '<span class="fa fa-'+ cat_classes[cat_label][1] +'">'+"</span>";
+    }else{
+    // Default button formatting
+        cat_label = '<span class="label label-default">'+ cat_label +"</span>";
+    }
+    return cat_label;
```

Comment:
I think the point of renaming `category` to `cat_label` was to avoid using the same variable for both things. So basically have a different variable name for whatever contains the html span element.